exporter/elasticsearchexporter/factory_test.go

```diff
@@ -94,3 +96,34 @@ func TestFactory_CreateLogsAndTracesExporterWithDeprecatedIndexOption(t *testing
 	require.NotNil(t, tracesExporter)
 	require.NoError(t, tracesExporter.Shutdown(context.Background()))
 }
+
+func TestFactory_DedotDeprecated(t *testing.T) {
+	factory := NewFactory()
+	cfg := withDefaultConfig(func(cfg *Config) {
+		cfg.Endpoint = "http://testing.invalid:9200"
+		cfg.Mapping.Dedot = false
+		cfg.Mapping.Mode = "ecs"
+	})
+
+	loggerCore, logObserver := observer.New(zap.WarnLevel)
+	set := exportertest.NewNopSettings()
+	set.Logger = zap.New(loggerCore)
+
+	logsExporter, err := factory.CreateLogsExporter(context.Background(), set, cfg)
+	require.NoError(t, err)
+	require.NoError(t, logsExporter.Shutdown(context.Background()))
+
+	tracesExporter, err := factory.CreateTracesExporter(context.Background(), set, cfg)
+	require.NoError(t, err)
+	require.NoError(t, tracesExporter.Shutdown(context.Background()))
+
+	metricsExporter, err := factory.CreateMetricsExporter(context.Background(), set, cfg)
+	require.NoError(t, err)
+	require.NoError(t, metricsExporter.Shutdown(context.Background()))
+
+	records := logObserver.AllUntimed()
+	assert.Len(t, records, 3)
+	assert.Equal(t, "dedot has been deprecated, and will always be enabled in ECS mode in future", records[0].Message)
+	assert.Equal(t, "dedot has been deprecated, and will always be enabled in ECS mode in future", records[1].Message)
+	assert.Equal(t, "dedot has been deprecated, and will always be enabled in ECS mode in future", records[2].Message)
```

Comment:
3 times is ok, maybe better if only once?

Reply:
It's once per exporter instantiation. Are you suggesting to only log the warning on the first instantiation? I'd rather not complicate this code, I intend to remove it pretty quickly.

Reply:
sorry I misunderstood. all ok!